File: ruby/strip_brackets.rb
#!/usr/bin/env ruby

require './utils.rb'

def strip_brackets(str)
  i = d = 0
  out = ''

  while i < str.length do
    d += 1 if str[i] == '{' || str[i] == '['
    out << if d > 0 then ' ' else str[i] end
    d -= 1 if str[i] == '}' || str[i] == ']'
    i += 1
  end

  out
end

ARGV.each do |f|
  t = benchmark { with_utf8_file(f) { |c| strip_brackets(c) } }
  puts "#{f}: #{t}"
end

File: python/strip_brackets.py
#!/usr/bin/env python

import utils, sys

def strip_brackets(string):
    i = d = 0
    out = ''
    while i < len(string):
        if string[i] == '{' or string[i] == '[':
            d += 1

        if d > 0:
            out += ' '
        else:
            out += string[i]

        if string[i] == '}' or string[i] == ']':
            d -= 1

        i += 1

    return out

for f in sys.argv[1:]:
    t = utils.benchmark(lambda: utils.with_utf8_file(f, strip_brackets))
    print('{0}: {1}'.format(f, t))
